packages/image_picker/image_picker_for_web/lib/image_resizer.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math';
+import 'dart:ui';
+import 'package:image_picker_for_web/image_resizer_utils.dart';
+import 'package:image_picker_platform_interface/image_picker_platform_interface.dart';
+import 'dart:html' as html;
+
+/// Helper class that resizes image
+class ImageResizer {
+  /// Resizes the image if needed
+  /// Does not support gif image
+  Future<XFile> resizeImageIfNeeded(XFile file, double? maxWidth,
+      double? maxHeight, int? imageQuality) async {
+    if (!imageResizeNeeded(maxWidth, maxHeight, imageQuality) ||
+        file.mimeType == "image/gif") {
+      //TODO Implement maxWidth and maxHeight for image/gif
+      return file;
+    }
+    try {
+      final imageElement = await loadImage(file.path);
+      final canvas = resizeImageElement(imageElement, maxWidth, maxHeight);
+      return writeCanvasToFile(file, canvas, imageQuality);
+    } catch (e) {
+      return file;
+    }
+  }
+
+  /// function that loads the blobUrl into an imageElement
+  Future<html.ImageElement> loadImage(String blobUrl) {
+    final imageLoadCompleter = Completer<html.ImageElement>();
+    final imageElement = html.ImageElement();
+    imageElement.src = blobUrl;
+
+    imageElement.onLoad.listen((event) {
+      html.Url.revokeObjectUrl(blobUrl);
```

Comment:
This is a potential bug. If we revokeObjectUrl here, and then `resizeImageElement` or `writeCanvasToFile` fail, we're returning a `file` object (line 28) that has a revoked blob URL, so its data can't be accessed.

Reply:
Forgot to push this change. done now.